openlibrary/catalog/add_book/__init__.py

```diff
@@ -676,52 +698,61 @@ def load_data(rec, account_key=None):
         edition['authors'] = authors
         reply['authors'] = author_reply
 
-    wkey = None
+    work_key = safeget(lambda: edition['works'][0]['key'])
     work_state = 'created'
     # Look for an existing work
-    if 'authors' in edition:
-        wkey = find_matching_work(edition)
-    if wkey:
-        w = web.ctx.site.get(wkey)
+    if not work_key and 'authors' in edition:
+        work_key = find_matching_work(edition)
+    if work_key:
+        work = web.ctx.site.get(work_key)
         work_state = 'matched'
-        found_wkey_match = True
         need_update = False
         for k in subject_fields:
             if k not in rec:
                 continue
             for s in rec[k]:
                 if normalize(s) not in [
-                    normalize(existing) for existing in w.get(k, [])
+                    normalize(existing) for existing in work.get(k, [])
                 ]:
-                    w.setdefault(k, []).append(s)
+                    work.setdefault(k, []).append(s)
                     need_update = True
         if cover_id:
-            w.setdefault('covers', []).append(cover_id)
+            work.setdefault('covers', []).append(cover_id)
             need_update = True
         if need_update:
             work_state = 'modified'
-            edits.append(w.dict())
+            edits.append(work.dict())
     else:
         # Create new work
-        w = new_work(edition, rec, cover_id)
-        wkey = w['key']
-        edits.append(w)
+        work = new_work(edition, rec, cover_id)
+        work_state = 'created'
+        work_key = work['key']
+        edits.append(work)
 
-    assert wkey
-    edition['works'] = [{'key': wkey}]
+    assert work_key
+    edition['works'] = [{'key': work_key}]
```

Comment:
```suggestion
    if not edition['works']:
        edition['works'] = [{'key': work_key}]
```

Not something we officially support, but this can have multiple items in it.

Reply:
I have made this change.